service/src/main/java/tds/config/repositories/impl/AccommodationsQueryRepositoryImpl.java

```diff
@@ -0,0 +1,187 @@
+package tds.config.repositories.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import tds.config.Accommodation;
+import tds.config.repositories.AccommodationsQueryRepository;
+
+@Repository
+public class AccommodationsQueryRepositoryImpl implements AccommodationsQueryRepository {
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    private static final String SEGMENTED_ASSESSMENT_SQL = "SELECT \n" +
+        "  distinct SegmentPosition as segment,\n" +
+        "  TType.DisableOnGuestSession as disableOnGuestSession, \n" +
+        "  TType.SortOrder as toolTypeSortOrder , \n" +
+        "  TT.SortOrder as toolValueSortOrder, \n" +
+        "  TType.TestMode as typeMode, \n" +
+        "  TT.TestMode as toolMode, \n" +
+        "  Type as accType, \n" +
+        "  Value as accValue, \n" +
+        "  Code as accCode, \n" +
+        "  IsDefault as isDefault, \n" +
+        "  AllowCombine as allowCombine, \n" +
+        "  IsFunctional as isFunctional, \n" +
+        "  AllowChange as allowChange,\n" +
+        "  IsSelectable as isSelectable, \n" +
+        "  IsVisible as isVisible, \n" +
+        "  studentControl as studentControl, \n" +
+        "  (select count(*) from configs.client_testtool TOOL where TOOL.ContextType = 'TEST' and TOOL.Context = MODE.testID and TOOL.clientname = MODE.clientname and TOOL.Type = TT.Type) as valCount, \n" +
+        "  null as dependsOnToolType, \n" +
+        "  IsEntryControl as isEntryControl\n" +
+        "FROM \n" +
+        "  configs.client_testmode MODE \n" +
+        "  JOIN configs.client_testtooltype TType ON \n" +
+        "    MODE.clientname = TType.clientname\n" +
+        "  JOIN configs.client_testtool TT ON \n" +
+        "    TT.Type = TType.Toolname AND \n" +
+        "    MODE.clientname = TT.clientname\n" +
+        "  JOIN configs.client_segmentproperties SEG ON \n" +
+        "    MODE.testkey = SEG.modekey AND \n" +
+        "    TType.Context = SEG.segmentID AND\n" +
+        "    TT.Context = SEG.segmentId\n" +
+        "WHERE \n" +
+        "  SEG.parentTest = MODE.testID \n" +
+        "  and MODE.testkey = :testKey \n" +
+        "  and TType.ContextType = 'SEGMENT' \n" +
+        "  and TT.ContextType = 'SEGMENT' \n" +
+        "  and (TType.TestMode = 'ALL' or TType.TestMode = MODE.mode) \n" +
+        "  and (TT.TestMode = 'ALL' or TT.TestMode = MODE.mode)";
```

Comment:
I'm guessing you got this from the original query so maybe it needs to stay like this, but I think there is a loophole in this logic.  It could match a record where in the TestTool table the TestMode is ALL and in the TestToolType table the TestMode = MODE.mode.  Probably unlikely, but I think the intent is to have the SQL be like this:


and (
	(TType.TestMode = 'ALL' and TT.TestMode = 'ALL')
	or
	(TT.TestMode = MODE.mode and TType.TestMode = MODE.mode)
)